syncopy/tests/helpers.py

```diff
@@ -0,0 +1,222 @@
+# -*- coding: utf-8 -*-
+#
+# Helper functions for frontend test design
+#
+# The runner signatures take a callable,
+# the `method_call`, as 1st argument
+#
+
+# 3rd party imports
+import itertools
+import numpy as np
+
+from syncopy.shared.errors import SPYValueError, SPYTypeError
+
```

Comment:
I understand and appreciate the way the `helpers` module reduces redundant test code by providing generic ways to test padding, `foi` etc. However, I'm not sure whether this additional layer of abstraction improves readability of the tests. For instance, the lambda construction in `test_gr_polyremoval` manages to cut down the testing code in the module to two lines (which is impressive!), but requires some non-trivial mental gymnastics to understand what's going on. That said, I'm quite aware that the dictionary for-loop festival in the selection tests isn't straight forward either. In the end it might be a matter of personal taste - maybe we can find some middle ground.

Reply:
I think in general it makes a lot of sense (DRY) to modularize standard tests (like foi, selection, polyremoval).. So once you understand the corresponding helper, you understood it for all existing and yet to come tests, which is a huge advantage imho. We can see how we improve the readability of said helpers?!

Reply:
Yup, let's do that. I'm still not convinced that DRY should be the law of the land when it comes to testing, though. The day will come when I need one of those helpers - I suggest continuing this discussion then.